### ISSUE TITLE
Muon Front End: NGPS PSU - Comms and configure on the Muon Front End [TIMEBOX: 1 day]

### ISSUE THREAD

**Issue by KathrynBaker** [ISISComputingGroup/IBEX#3447]
As a muon front end user I want to be able to control the NGPS/OCEM PSUs from NDEMUONFE

**Acceptance Criteria**
1. The IOCs are able to connect to the PSUs and read and set/commands as per #1922

**Notes**
1. Rory and Kathryn have seen a PSU respond - this has only been successful so far by running in a 'local' network mode - Laptop set to a 192.168 address same as the PSU, the serial connection has not been successful so far
1. The PSUs will be installed with Cat cables - terminated as standard running from the vicinity of the MOXAs to the PSUs, with enough length to be used at the back (serial) or the front (ethernet)
1. It is possible that the serial will function with the 422/485 setup of the MOXA done properly, the magnet team are going to chase up what the serial settings should be
1. It may be possible to run them on the R55 network, again the supplier will be asked about DHCP - it may be that on the open network they are unstable - this cannot be tested until they are in situ
1. It may be necessary to run them on the network via a firewall to overcome the previous issue if they have to run in a local mode
1. The internal IOC was not tested, and the information is again being sought from the supplier. Rory was able to talk to a PSU via PuTTY so we do know that a homegrown IOC will work
1. We have asked for 3 days to get these devices sorted in the end

**Comment by KathrynBaker** [comment]
I am putting this straight into awaiting, as this may come up in the next two weeks, or given some of the hardware issues that have been had, the install of these PSUs may be delayed until the end of the calendar year (December 2018/January 2019)

**Comment by KathrynBaker** [comment]
This may be brought forward to the October 2018 shutdown

**Comment by Tom-Willemsen** [comment]
Tried with a wide variety of different pinouts by using a small adapter (which was in the box with the serial sniffer) that lets you easily change pinouts. I couldn't find any combination which worked from the moxa.

What I tried:
- The adapter that @KathrynBaker made up
  * Against the different settings in the moxa (422/485 2-wire/485 4-wire)
  * With all the permutations of data lines being swapped/polarities reversed
  * With and without null modems in each case
- Two adapters that @FreddieAkeroyd and I made up yesterday evening
  * Against the different moxa settings
  * With and without null modems

None of these approaches got me any data back from the PSU. However, talking to the PSU from my laptop using the USB-RS422/485 converter *does* let me talk.

Given that this ticket was timeboxed to 1 day, and that I have now spent a day attempting to get serial to work via the moxa, I think we should try the ethernet option, with a firewall box.
